spec/requests/api/provision_requests_spec.rb

```diff
@@ -105,6 +105,140 @@ def app
     end
   end
 
+  context "AWS advanced provision requests" do
+    let(:aws_dialog)     { FactoryGirl.create(:miq_dialog_aws_provision) }
+    let(:ems)            { FactoryGirl.create(:ems_amazon_with_authentication) }
+    let(:template) do
+      FactoryGirl.create(:template_amazon, :name => "template1", :ext_management_system => ems)
+    end
+    let(:flavor) do
+      FactoryGirl.create(:flavor_amazon, :ems_id => ems.id, :name => 't2.small', :cloud_subnet_required => true)
+    end
+    let(:az)             { FactoryGirl.create(:availability_zone_amazon) }
+    let(:cloud_network1) { FactoryGirl.create(:cloud_network, :ems_id => ems.id, :enabled => true) }
+    let(:cloud_subnet1)  { FactoryGirl.create(:cloud_subnet, :cloud_network_id => cloud_network1.id) }
```

Comment:
These resources need a relationship to the provider (ems) and the subnet also needs a relationship to the availability_zone.
```ruby
let(:az)             { FactoryGirl.create(:availability_zone_amazon, :ems_id => ems.id) }
let(:cloud_network1) { FactoryGirl.create(:cloud_network, :ems_id => ems.id, :enabled => true) }
let(:cloud_subnet1) do
  FactoryGirl.create(:cloud_subnet, :ems_id => ems.id, :cloud_network => cloud_network1, :availability_zone => az)
end
```